src/pip/_internal/commands/show.py

```diff
@@ -166,3 +182,83 @@ def print_results(distributions, list_files=False, verbose=False):
             if "files" not in dist:
                 logger.info("Cannot locate installed-files.txt")
     return results_printed
+
+
+def print_json(distributions, list_files=False, verbose=False):
+    """
+    Print in JSON format the information from installed distributions found.
+    """
+    all_results = []
+
+    results_printed = False
+    for i, dist in enumerate(distributions):
+        results_printed = True
+
+        name = dist.get('name', '')
+        required_by = [
+            pkg.project_name for pkg in pkg_resources.working_set
+            if name in [required.name for required in pkg.requires()]
+        ]
+
+        results = [
+            ("Name", name),
+            ("Version", dist.get('version', '')),
+            ("Summary", dist.get('summary', '')),
+            ("HomePage", dist.get('home-page', '')),
+            ("Author", dist.get('author', '')),
+            ("AuthorEmail", dist.get('author-email', '')),
+            ("License", dist.get('license', '')),
+            ("Location", dist.get('location', '')),
+            ("Requires", ','.join(dist.get('requires', [])).split(',')),
+            ("RequiredBy", ','.join(required_by).split(','))
+        ]
+
+        if verbose:
+            classifiers = []
+            for classifier in dist.get('classifiers', []):
+                classifiers.append(str(classifier))
+
+            # 2.7/3.x compatibility
+            class ReadlineWrapper:
```

Comment:
Why the added complexity vs what is used in `print_header_format` ?

Reply:
`print_header_format` prints line by line and here we need to parse the distribution and convert it into json.  For entry_points there is a generator object that we also need to parse. The config parser for 3.6 handles this no problem, but not 2.7 it needs extra help:
{'name': 'pytest', 'version': '3.8.2', 'location': '/Users/karadeyoungdelamarck/Documents/pip/.tox/py36/lib/python3.6/site-packages', 'requires': ['atomicwrites', 'py', 'six', 'setuptools', 'pluggy', 'attrs', 'more-itertools'], 'entry_points': <generator object yield_lines at 0x10f0e6620>, 'installer': 'pip', 'metadata-version': '2.1',

Reply:
But I'd be happy to have suggestions on a simpler way to handle this.

Does this project (pypa or pip) have a slack or gitter channel for questions?